src/main/java/core/basesyntax/dao/impl/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,218 @@
+                + "FROM cars "
+                + "Inner join manufacturers "
+                + "ON cars.manufacturer_id = manufacturers.manufacturer_id "
+                + "Where cars.deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            return getCars(resultSet);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars ", e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ?"
+                + " WHERE car_id = ? AND deleted = false ";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.setLong(3, car.getId());
+            preparedStatement.executeUpdate();
+            deleteDriversForCar(car.getId());
+            addDriversForCar(car);
+            int updateRows = preparedStatement.executeUpdate();
+            if (updateRows > 0) {
+                return car;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update this car " + car, e);
+        }
+        throw new DataProcessingException("No car with id " + car.getId());
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET deleted = true WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            deleteDriversForCar(id);
+            int updateRows = preparedStatement.executeUpdate();
+            return updateRows > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = " SELECT c.car_id as car_id,"
+                + " c.model as model,"
+                + " c.manufacturer_id as manufacturer_id,"
+                + " m.manufacturer_country as manufacturer_country, "
+                + " m.manufacturer_name as manufacturer_name "
+                + "FROM cars_drivers cd "
+                + "INNER JOIN cars c "
+                + "ON cd.car_id = c.car_id "
+                + "INNER JOIN manufacturers m "
+                + "ON c.manufacturer_id = m.manufacturer_id "
+                + "WHERE cd.driver_id = ? AND c.deleted = false";
```

Comment:
let’s join `drivers` table as well - we shouldn’t fetch info if the driver is deleted